cdap-ui/app/cdap/api/__mocks__/dataset.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+import Rx from 'rx';
+const MyDatasetApi = {
+  __list: []
+};
+
+MyDatasetApi.list = function() {
+  let subject = new Rx.Subject();
+  setTimeout(() => {
+    subject.onNext(this.__list);
+  });
+  return subject;
+};
+
+export {MyDatasetApi};
```

Comment:
Not sure why you're doing `export {}` in some places, `export default` in some others, as well as `module.exports = {}`.